engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterPlatformViewsController.mm

```diff
@@ -665,8 +665,23 @@ - (BOOL)submitFrame:(std::unique_ptr<flutter::SurfaceFrame>)background_frame
      withIosContext:(const std::shared_ptr<flutter::IOSContext>&)iosContext {
   TRACE_EVENT0("flutter", "PlatformViewsController::SubmitFrame");
 
-  // No platform views to render; we're done.
+  // No platform views to render.
   if (self.flutterView == nil || (self.compositionOrder.empty() && !self.hadPlatformViews)) {
+    // No platform views to render but the FlutterView may need to be resized.
+    if (self.flutterView != nil) {
+      // If the raster thread isn't merged, resize the view on the platform thread and block until
+      // complete.
+      auto latch = std::make_shared<fml::CountDownLatch>(1u);
+      fml::TaskRunner::RunNowOrPostTask(self.platformTaskRunner,
+                                        [self, frameSize = self.frameSize, latch]() mutable {
+                                          [self performResize:frameSize];
+                                          latch->CountDown();
+                                        });
+      if (![[NSThread currentThread] isMainThread]) {
+        latch->Wait();
+      }
```

Comment:
Can we just not make the latch if we are on the platform thread?  Can you also unify the calls for threading by using TaskRunner::RunsTasksOnCurrentThread.

That will allow you to also get rid of the shared_ptr around the latch.